SignalMessaging/environment/OWSAudioSession.swift

```diff
@@ -84,6 +90,12 @@ public class OWSAudioSession: NSObject {
 
         startAudioActivity(audioActivity)
 
+        guard currentActivities.count == 1 else {
```

Comment:
I'm not sure this is right.

To simplify a bit:

We want to allow "upgrading" audio sessions:

Ambient -> Playback -> PlayAndRecord

But we never want to "downgrade" functionality.

E.g. If we are playing a video while a call comes in, we *want* to go from Playback -> PlayAndRecord

We just want to make sure the reverse doesn't happen - while on a call, when we play a video, we don't want to go from "PlayAndRecord" -> "Playback".

And a disclaimer: I'm not sure that a linear progression is nuanced enough to capture all the various transitions we want to make to our audio session configuration, but I'm pretty sure it would be better than what we have.

Reply:
And there's theoretically a similar issue with downgrading from Playback -> Ambient, but off the top of my head, I'm not sure where we use ambient, so it might not be an issue in practice.

The reason the guard exists in ambient right now is that ambient is the "most restrictive" (e.g. respects silent switch, doesn't require recording permission). 

So if we're starting an ambient activity, and any other activity exists, that activity has either already configured the audio session to ambient which is a no-op or either "Playback" or "PlaybackAndRecord", which should suffice for any ambient audio activity.

Reply:
Okay, let's address this while stabilizing v2.27.